src/common/map_locations.h

```diff
@@ -0,0 +1,92 @@
+/*
+    This file is part of darktable,
+    Copyright (C) 2010-2020 darktable developers.
+
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <glib.h>
+#include <sqlite3.h>
+#include <stdint.h>
+
+
+typedef enum dt_map_locations_type_t
+{
+  MAP_LOCATION_SHAPE_CIRCLE,
+  MAP_LOCATION_SHAPE_RECTANGLE,
+  MAP_LOCATION_SHAPE_FORM_MAX
+} dt_map_locations_type_t;
+
+typedef struct dt_map_location_data_t
+{
+  double lon, lat, delta1, delta2;
+  int shape;
+} dt_map_location_data_t;
+
+typedef struct dt_map_location_t
+{
+  guint id;
+  gchar *tag;
+  guint count;
+} dt_map_location_t;
+
+// create a new location
+const guint dt_map_location_new(const char *const name);
+
+// remove a location
+void dt_map_location_delete(const guint tagid);
```

Comment:
As a clean-up I would also rename all tagid in the parameters to locid. I understand that now the relation between both but it reads better here to talk about location id and not tag id.

Reply:
Ok, so to resume, on db side, back to tagid, and all location parameter usage of locid instead of tagid, right ?

Reply:
Right, that's my suggestion. Is that ok with you ?

Reply:
yes